robot_smach_states/src/robot_smach_states/navigation/navigate_to_designator.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import absolute_import
+
+# TU/e Robotics
+from .navigation import NavigateTo
+from ..util.designators import check_resolve_type
+
+
+class NavigateToDesignator(NavigateTo):
+    def __init__(self, robot, constraint_designator, reset_head=True, speak=True):
+        """
+        @param constraint_designator a Designator that resolves to navigation constraints
+        """
+        super(NavigateToDesignator, self).__init__(robot, reset_head=reset_head, speak=speak)
+
+        check_resolve_type(constraint_designator, tuple)  #TODO verify in more detail that this is indeed a constraint designator
```

Comment:
@LoyVanBeek As designator expert, do you think there is a way to verify that the constraint resolves to a tuple containing a Positionconstraint and an Orientationconstraints?

Reply:
I don't think so, as tuples are interpreted as something else.
But the neater way to do this might be to return a custom type, eg. a `PoseConstraint` which is composed of both a `PositionConstraint` and a `OrientationConstraint`. 
Will also clean up some type hinting.